scipy/signal/signaltools.py

```diff
@@ -463,6 +460,9 @@ def convolve(in1, in2, mode='full'):
     volume = asarray(in1)
     kernel = asarray(in2)
 
+    if not _is_valid_mode_shapes(mode, volume.shape, kernel.shape):
+        volume, kernel = kernel, volume
```

Comment:
actually it's faster to just work with tuples directly:

```
    def _check_valid_mode_shapes(shape1, shape2):
        ok1, ok2 = True, True
        for d1, d2 in zip(shape1, shape2):
            if not d1 >= d2:
                ok1 = False
            if not d2 >= d1:
                ok2 = False
        if not (ok1 or ok2):
            raise ValueError(
                    "For 'valid' mode, one array must be smaller "
                    "than the other in every dimension.")
```

this takes about 2 μs.  the np.maximum version took 30 μs.  for comparison, `np.convolve(2, 3)` takes 10 µs.